libvast/test/meta_index.cpp

```diff
@@ -303,4 +303,13 @@ TEST(serialization with custom factory) {
   CHECK_ROUNDTRIP(meta_idx);
 }
 
+TEST(option setting and retrieval) {
+  meta_index meta_idx;
+  auto& opts = meta_idx.factory_options();
+  put(opts, "foo", 42);
+  auto x = caf::get_if<caf::config_value::integer>(&opts["foo"]);
+  REQUIRE(x);
```

Comment:
For future reference, I think we should always use `CHECK_EQUAL(unbox(x), 42)` instead to reduce unnecessary boilerplate-checks.

Reply:
Absolutely; `unbox` is just a bit new, and the pervasive `REQUIRE-CHECK` pattern takes a while to unlearn.

Reply:
So I just wanted to replace this in master to get used to unbox, but the compiler complains with your suggestion:
```
.../caf/libcaf_test/caf/test/dsl.hpp:776:3: note: candidate template ignored: could not match 'expected<type-parameter-0-0>' against 'long long *'
T unbox(caf::expected<T> x) {
  ^
.../caf/libcaf_test/caf/test/dsl.hpp:784:3: note: candidate template ignored: could not match 'optional<type-parameter-0-0>' against 'long long *'
T unbox(caf::optional<T> x) {
```
Should we add an overload of `unbox` for pointers?

Reply:
I'd suggest this patch to CAF:
```diff
diff --git a/libcaf_test/caf/test/dsl.hpp b/libcaf_test/caf/test/dsl.hpp
index 0a9ddaa4..8741407c 100644
--- a/libcaf_test/caf/test/dsl.hpp
+++ b/libcaf_test/caf/test/dsl.hpp
@@ -787,6 +787,14 @@ T unbox(caf::optional<T> x) {
   return std::move(*x);
 }

+/// Unboxes an optional value or fails the test if it doesn't exist.
+template <class T>
+T unbox(T* x) {
+  if (x == nullptr)
+    CAF_FAIL("x == nullptr");
+  return std::move(*x);
+}
+
 /// Expands to its argument.
 #define CAF_EXPAND(x) x
```

Reply:
Sure, I don't see a reason why not: https://github.com/actor-framework/actor-framework/commit/0b9b191ec6c5615b4a4b32d94ddf09e654f8b8e0.

Reply:
Ah, didn't see your patch before pushing. `move(*x)` could cause serious trouble, because we aren't owning what's in `x`. For example, we could silently move a value out something by accident that we accessed via `get_if`.

Reply:
But isn't that then the same problem with the `expected` and `optional` overloads?

Reply:
No, `expected` and `optional` both own their content. Moving their content away is fine, because we destroy the temporary function argument at scope exit.